In [1]:
from google.colab import drive

In [2]:
drive.mount('/content/drive')

Mounted at /content/drive


#  **EDA - Unnesting Datasets 1 and 3**

We have two other .json files that need to be explored and unnested:

- steam_games.json
- users_items.json

### 1. Data Load:

### *1.1. Steam_games Dataset:*

In [3]:
import pandas as pd
import json

data = []
with open('/content/drive/MyDrive/Colab Notebooks/Dataset/steam_games.json', 'r') as file:
    for line in file:
        data.append(json.loads(line))

df_steam_games = pd.DataFrame(data)
df_steam_games

publisher                                 genres  \
0                   NaN                                    NaN   
1                   NaN                                    NaN   
2                   NaN                                    NaN   
3                   NaN                                    NaN   
4                   NaN                                    NaN   
...                 ...                                    ...   
120440  Ghost_RUS Games  [Casual, Indie, Simulation, Strategy]   
120441           Sacada              [Casual, Indie, Strategy]   
120442     Laush Studio            [Indie, Racing, Simulation]   
120443         SIXNAILS                        [Casual, Indie]   
120444              NaN                                    NaN   

                        app_name                     title  \
0                            NaN                       NaN   
1                            NaN                       NaN   
2                            NaN                       NaN   
3                            NaN                       NaN   
4                            NaN                       NaN   
...                          ...                       ...   
120440            Colony On Mars            Colony On Mars   
120441  LOGistICAL: South Africa  LOGistICAL: South Africa   
120442             Russian Roads             Russian Roads   
120443       EXIT 2 - Directions       EXIT 2 - Directions   
120444               Maze Run VR                       NaN   

                                                      url release_date  \
0                                                     NaN          NaN   
1                                                     NaN          NaN   
2                                                     NaN          NaN   
3                                                     NaN          NaN   
4                                                     NaN          NaN   
...                                                   ...          ...   
120440  http://store.steampowered.com/app/773640/Colon...   2018-01-04   
120441  http://store.steampowered.com/app/733530/LOGis...   2018-01-04   
120442  http://store.steampowered.com/app/610660/Russi...   2018-01-04   
120443  http://store.steampowered.com/app/658870/EXIT_...   2017-09-02   
120444  http://store.steampowered.com/app/681550/Maze_...          NaN   

                                                     tags  \
0                                                     NaN   
1                                                     NaN   
2                                                     NaN   
3                                                     NaN   
4                                                     NaN   
...                                                   ...   
120440              [Strategy, Indie, Casual, Simulation]   
120441                          [Strategy, Indie, Casual]   
120442                        [Indie, Simulation, Racing]   
120443  [Indie, Casual, Puzzle, Singleplayer, Atmosphe...   
120444  [Early Access, Adventure, Indie, Action, Simul...   

                                              reviews_url  \
0                                                     NaN   
1                                                     NaN   
2                                                     NaN   
3                                                     NaN   
4                                                     NaN   
...                                                   ...   
120440  http://steamcommunity.com/app/773640/reviews/?...   
120441  http://steamcommunity.com/app/733530/reviews/?...   
120442  http://steamcommunity.com/app/610660/reviews/?...   
120443  http://steamcommunity.com/app/658870/reviews/?...   
120444  http://steamcommunity.com/app/681550/reviews/?...   

                                                    specs price early_access  \
0                                        

In [4]:
df_steam_games.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 120445 entries, 0 to 120444
Data columns (total 13 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   publisher     24083 non-null  object
 1   genres        28852 non-null  object
 2   app_name      32133 non-null  object
 3   title         30085 non-null  object
 4   url           32135 non-null  object
 5   release_date  30068 non-null  object
 6   tags          31972 non-null  object
 7   reviews_url   32133 non-null  object
 8   specs         31465 non-null  object
 9   price         30758 non-null  object
 10  early_access  32135 non-null  object
 11  id            32133 non-null  object
 12  developer     28836 non-null  object
dtypes: object(13)
memory usage: 11.9+ MB


### *1.2. Users_items Dataset:*

In [5]:
# Due to the file users_items.json has some dictionaries and list nested into a table that is not a .json format,
# we may need to use the library "ast". The ast.literal_eval() function is capable of evaluating Python literal
# expressions, including dictionaries and lists. By reading each line of the file and applying ast.literal_eval(),
# you are interpreting each line as a Python literal expression and converting it into a Python object.

import ast

data = []
with open('/content/drive/MyDrive/Colab Notebooks/Dataset/users_items.json', 'r', encoding='utf-8') as file:
    for line in file.readlines():
        data.append(ast.literal_eval(line))

df_users_items = pd.DataFrame(data)
df_users_items

user_id  items_count           steam_id  \
0            76561197970982479          277  76561197970982479   
1                      js41637          888  76561198035864385   
2                    evcentric          137  76561198007712555   
3                   Riot-Punch          328  76561197963445855   
4                        doctr          541  76561198002099482   
...                        ...          ...                ...   
88305        76561198323066619           22  76561198323066619   
88306        76561198326700687          177  76561198326700687   
88307  XxLaughingJackClown77xX            0  76561198328759259   
88308        76561198329548331            7  76561198329548331   
88309        edward_tremethick            0  76561198331598578   

                                                user_url  \
0      http://steamcommunity.com/profiles/76561197970...   
1                   http://steamcommunity.com/id/js41637   
2                 http://steamcommunity.com/id/evcentric   
3                http://steamcommunity.com/id/Riot-Punch   
4                     http://steamcommunity.com/id/doctr   
...                                                  ...   
88305  http://steamcommunity.com/profiles/76561198323...   
88306  http://steamcommunity.com/profiles/76561198326...   
88307  http://steamcommunity.com/id/XxLaughingJackClo...   
88308  http://steamcommunity.com/profiles/76561198329...   
88309     http://steamcommunity.com/id/edward_tremethick   

                                                   items  
0      [{'item_id': '10', 'item_name': 'Counter-Strik...  
1      [{'item_id': '10', 'item_name': 'Counter-Strik...  
2      [{'item_id': '1200', 'item_name': 'Red Orchest...  
3      [{'item_id': '10', 'item_name': 'Counter-Strik...  
4      [{'item_id': '300', 'item_name': 'Day of Defea...  
...                                                  ...  
88305  [{'item_id': '413850', 'item_name': 'CS:GO Pla...  
88306  [{'item_id': '11020', 'item_name': 'TrackMania...  
88307                                                 []  
88308  [{'item_id': '304930', 'item_name': 'Unturned'...  
88309                                                 []  

[88310 rows x 5 columns]

In [6]:
df_users_items.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 88310 entries, 0 to 88309
Data columns (total 5 columns):
 #   Column       Non-Null Count  Dtype 
---  ------       --------------  ----- 
 0   user_id      88310 non-null  object
 1   items_count  88310 non-null  int64 
 2   steam_id     88310 non-null  object
 3   user_url     88310 non-null  object
 4   items        88310 non-null  object
dtypes: int64(1), object(4)
memory usage: 3.4+ MB


### 2. Data Transformation:

### *2.1. Steam_games Dataset:*

We can see that in the df_steam_games we have 120,445 entries, but the first rows are null in every column. So it would be a good idea to eliminate them:

In [7]:
df_steam_games1 = df_steam_games.copy()
df_steam_games1

publisher                                 genres  \
0                   NaN                                    NaN   
1                   NaN                                    NaN   
2                   NaN                                    NaN   
3                   NaN                                    NaN   
4                   NaN                                    NaN   
...                 ...                                    ...   
120440  Ghost_RUS Games  [Casual, Indie, Simulation, Strategy]   
120441           Sacada              [Casual, Indie, Strategy]   
120442     Laush Studio            [Indie, Racing, Simulation]   
120443         SIXNAILS                        [Casual, Indie]   
120444              NaN                                    NaN   

                        app_name                     title  \
0                            NaN                       NaN   
1                            NaN                       NaN   
2                            NaN                       NaN   
3                            NaN                       NaN   
4                            NaN                       NaN   
...                          ...                       ...   
120440            Colony On Mars            Colony On Mars   
120441  LOGistICAL: South Africa  LOGistICAL: South Africa   
120442             Russian Roads             Russian Roads   
120443       EXIT 2 - Directions       EXIT 2 - Directions   
120444               Maze Run VR                       NaN   

                                                      url release_date  \
0                                                     NaN          NaN   
1                                                     NaN          NaN   
2                                                     NaN          NaN   
3                                                     NaN          NaN   
4                                                     NaN          NaN   
...                                                   ...          ...   
120440  http://store.steampowered.com/app/773640/Colon...   2018-01-04   
120441  http://store.steampowered.com/app/733530/LOGis...   2018-01-04   
120442  http://store.steampowered.com/app/610660/Russi...   2018-01-04   
120443  http://store.steampowered.com/app/658870/EXIT_...   2017-09-02   
120444  http://store.steampowered.com/app/681550/Maze_...          NaN   

                                                     tags  \
0                                                     NaN   
1                                                     NaN   
2                                                     NaN   
3                                                     NaN   
4                                                     NaN   
...                                                   ...   
120440              [Strategy, Indie, Casual, Simulation]   
120441                          [Strategy, Indie, Casual]   
120442                        [Indie, Simulation, Racing]   
120443  [Indie, Casual, Puzzle, Singleplayer, Atmosphe...   
120444  [Early Access, Adventure, Indie, Action, Simul...   

                                              reviews_url  \
0                                                     NaN   
1                                                     NaN   
2                                                     NaN   
3                                                     NaN   
4                                                     NaN   
...                                                   ...   
120440  http://steamcommunity.com/app/773640/reviews/?...   
120441  http://steamcommunity.com/app/733530/reviews/?...   
120442  http://steamcommunity.com/app/610660/reviews/?...   
120443  http://steamcommunity.com/app/658870/reviews/?...   
120444  http://steamcommunity.com/app/681550/reviews/?...   

                                                    specs price early_access  \
0                                        

In [8]:
# Mask creation

mask1 = df_steam_games['publisher'].isna() & df_steam_games['genres'].isna() & df_steam_games['app_name'].isna() & df_steam_games['title'].isna() & df_steam_games['url'].isna() & df_steam_games['release_date'].isna() & df_steam_games['tags'].isna() & df_steam_games['reviews_url'].isna() & df_steam_games['specs'].isna() & df_steam_games['price'].isna() & df_steam_games['early_access'].isna() & df_steam_games['id'].isna() & df_steam_games['developer'].isna()
df_steam_games[mask1]

publisher genres app_name title  url release_date tags reviews_url  \
0           NaN    NaN      NaN   NaN  NaN          NaN  NaN         NaN   
1           NaN    NaN      NaN   NaN  NaN          NaN  NaN         NaN   
2           NaN    NaN      NaN   NaN  NaN          NaN  NaN         NaN   
3           NaN    NaN      NaN   NaN  NaN          NaN  NaN         NaN   
4           NaN    NaN      NaN   NaN  NaN          NaN  NaN         NaN   
...         ...    ...      ...   ...  ...          ...  ...         ...   
88305       NaN    NaN      NaN   NaN  NaN          NaN  NaN         NaN   
88306       NaN    NaN      NaN   NaN  NaN          NaN  NaN         NaN   
88307       NaN    NaN      NaN   NaN  NaN          NaN  NaN         NaN   
88308       NaN    NaN      NaN   NaN  NaN          NaN  NaN         NaN   
88309       NaN    NaN      NaN   NaN  NaN          NaN  NaN         NaN   

      specs price early_access   id developer  
0       NaN   NaN          NaN  NaN       NaN  
1       NaN   NaN          NaN  NaN       NaN  
2       NaN   NaN          NaN  NaN       NaN  
3       NaN   NaN          NaN  NaN       NaN  
4       NaN   NaN          NaN  NaN       NaN  
...     ...   ...          ...  ...       ...  
88305   NaN   NaN          NaN  NaN       NaN  
88306   NaN   NaN          NaN  NaN       NaN  
88307   NaN   NaN          NaN  NaN       NaN  
88308   NaN   NaN          NaN  NaN       NaN  
88309   NaN   NaN          NaN  NaN       NaN  

[88310 rows x 13 columns]

We can see that those entries that have all its columns with NaN are the first 88310 (from row 0 to row 88309). Let's eliminate them from te dataframe.

In [9]:
df_steam_games1.drop(df_steam_games1.index[0:88310], inplace=True)
df_steam_games1

publisher                                             genres  \
88310          Kotoshiro      [Action, Casual, Indie, Simulation, Strategy]   
88311   Making Fun, Inc.               [Free to Play, Indie, RPG, Strategy]   
88312       Poolians.com  [Casual, Free to Play, Indie, Simulation, Sports]   
88313               彼岸领域                        [Action, Adventure, Casual]   
88314                NaN                                                NaN   
...                  ...                                                ...   
120440   Ghost_RUS Games              [Casual, Indie, Simulation, Strategy]   
120441            Sacada                          [Casual, Indie, Strategy]   
120442      Laush Studio                        [Indie, Racing, Simulation]   
120443          SIXNAILS                                    [Casual, Indie]   
120444               NaN                                                NaN   

                        app_name                     title  \
88310        Lost Summoner Kitty       Lost Summoner Kitty   
88311                  Ironbound                 Ironbound   
88312    Real Pool 3D - Poolians   Real Pool 3D - Poolians   
88313                    弹炸人2222                   弹炸人2222   
88314              Log Challenge                       NaN   
...                          ...                       ...   
120440            Colony On Mars            Colony On Mars   
120441  LOGistICAL: South Africa  LOGistICAL: South Africa   
120442             Russian Roads             Russian Roads   
120443       EXIT 2 - Directions       EXIT 2 - Directions   
120444               Maze Run VR                       NaN   

                                                      url release_date  \
88310   http://store.steampowered.com/app/761140/Lost_...   2018-01-04   
88311   http://store.steampowered.com/app/643980/Ironb...   2018-01-04   
88312   http://store.steampowered.com/app/670290/Real_...   2017-07-24   
88313      http://store.steampowered.com/app/767400/2222/   2017-12-07   
88314   http://store.steampowered.com/app/773570/Log_C...          NaN   
...                                                   ...          ...   
120440  http://store.steampowered.com/app/773640/Colon...   2018-01-04   
120441  http://store.steampowered.com/app/733530/LOGis...   2018-01-04   
120442  http://store.steampowered.com/app/610660/Russi...   2018-01-04   
120443  http://store.steampowered.com/app/658870/EXIT_...   2017-09-02   
120444  http://store.steampowered.com/app/681550/Maze_...          NaN   

                                                     tags  \
88310       [Strategy, Action, Indie, Casual, Simulation]   
88311   [Free to Play, Strategy, Indie, RPG, Card Game...   
88312   [Free to Play, Simulation, Sports, Casual, Ind...   
88313                         [Action, Adventure, Casual]   
88314                     [Action, Indie, Casual, Sports]   
...                                                   ...   
120440              [Strategy, Indie, Casual, Simulation]   
120441                          [Strategy, Indie, Casual]   
120442                        [Indie, Simulation, Racing]   
120443  [Indie, Casual, Puzzle, Singleplayer, Atmosphe...   
120444  [Early Access, Adventure, Indie, Action, Simul...   

                                              reviews_url  \
88310   http://steamcommunity.com/app/761140/reviews/?...   
88311   http://steamcommunity.com/app/643980/reviews/?...   
88312   http://steamcommunity.com/app/670290/reviews/?...   
88313   http://steamcommunity.com/app/767400/reviews/?...   
88314   http://steamcommunity.com/app/773570/reviews/?...   
...                                                   ...   
120440  http://steamcommunity.com/app/773640/reviews/?...   
120441  http://steamcommunity.com/app/733530/reviews/?...   
120442  http://steamcommunity.com/app/610660/reviews/?...   
120443  http://steamcommunity.com/app/658870/reviews/?...   
120444  http://steamcommunit

We have now 32135 entries. Now let's reset the index of this dataframe.

In [10]:
df_steam_games1.reset_index(drop=True, inplace=True)
df_steam_games1

publisher                                             genres  \
0             Kotoshiro      [Action, Casual, Indie, Simulation, Strategy]   
1      Making Fun, Inc.               [Free to Play, Indie, RPG, Strategy]   
2          Poolians.com  [Casual, Free to Play, Indie, Simulation, Sports]   
3                  彼岸领域                        [Action, Adventure, Casual]   
4                   NaN                                                NaN   
...                 ...                                                ...   
32130   Ghost_RUS Games              [Casual, Indie, Simulation, Strategy]   
32131            Sacada                          [Casual, Indie, Strategy]   
32132      Laush Studio                        [Indie, Racing, Simulation]   
32133          SIXNAILS                                    [Casual, Indie]   
32134               NaN                                                NaN   

                       app_name                     title  \
0           Lost Summoner Kitty       Lost Summoner Kitty   
1                     Ironbound                 Ironbound   
2       Real Pool 3D - Poolians   Real Pool 3D - Poolians   
3                       弹炸人2222                   弹炸人2222   
4                 Log Challenge                       NaN   
...                         ...                       ...   
32130            Colony On Mars            Colony On Mars   
32131  LOGistICAL: South Africa  LOGistICAL: South Africa   
32132             Russian Roads             Russian Roads   
32133       EXIT 2 - Directions       EXIT 2 - Directions   
32134               Maze Run VR                       NaN   

                                                     url release_date  \
0      http://store.steampowered.com/app/761140/Lost_...   2018-01-04   
1      http://store.steampowered.com/app/643980/Ironb...   2018-01-04   
2      http://store.steampowered.com/app/670290/Real_...   2017-07-24   
3         http://store.steampowered.com/app/767400/2222/   2017-12-07   
4      http://store.steampowered.com/app/773570/Log_C...          NaN   
...                                                  ...          ...   
32130  http://store.steampowered.com/app/773640/Colon...   2018-01-04   
32131  http://store.steampowered.com/app/733530/LOGis...   2018-01-04   
32132  http://store.steampowered.com/app/610660/Russi...   2018-01-04   
32133  http://store.steampowered.com/app/658870/EXIT_...   2017-09-02   
32134  http://store.steampowered.com/app/681550/Maze_...          NaN   

                                                    tags  \
0          [Strategy, Action, Indie, Casual, Simulation]   
1      [Free to Play, Strategy, Indie, RPG, Card Game...   
2      [Free to Play, Simulation, Sports, Casual, Ind...   
3                            [Action, Adventure, Casual]   
4                        [Action, Indie, Casual, Sports]   
...                                                  ...   
32130              [Strategy, Indie, Casual, Simulation]   
32131                          [Strategy, Indie, Casual]   
32132                        [Indie, Simulation, Racing]   
32133  [Indie, Casual, Puzzle, Singleplayer, Atmosphe...   
32134  [Early Access, Adventure, Indie, Action, Simul...   

                                             reviews_url  \
0      http://steamcommunity.com/app/761140/reviews/?...   
1      http://steamcommunity.com/app/643980/reviews/?...   
2      http://steamcommunity.com/app/670290/reviews/?...   
3      http://steamcommunity.com/app/767400/reviews/?...   
4      http://steamcommunity.com/app/773570/reviews/?...   
...                                                  ...   
32130  http://steamcommunity.com/app/773640/reviews/?...   
32131  http://steamcommunity.com/app/733530/reviews/?...   
32132  http://steamcommunity.com/app/610660/reviews/?...   
32133  http://steamcommunity.com/app/658870/reviews/?...   
32134  http://steamcommunity.com/app/681550/reviews/?...   

                         

Now, that the dataframe is organized cleaned from the the entries without information, we are going to proceed to load it into a CSV file as follows:

In [15]:
df_steam_games1.to_csv('steam_games_final.csv', index=False)

In [18]:
dfPrueba1 = pd.read_csv('/content/drive/MyDrive/Colab Notebooks/steam_games_final.csv')
dfPrueba1

publisher                                             genres  \
0             Kotoshiro  ['Action', 'Casual', 'Indie', 'Simulation', 'S...   
1      Making Fun, Inc.       ['Free to Play', 'Indie', 'RPG', 'Strategy']   
2          Poolians.com  ['Casual', 'Free to Play', 'Indie', 'Simulatio...   
3                  彼岸领域                  ['Action', 'Adventure', 'Casual']   
4                   NaN                                                NaN   
...                 ...                                                ...   
32130   Ghost_RUS Games      ['Casual', 'Indie', 'Simulation', 'Strategy']   
32131            Sacada                    ['Casual', 'Indie', 'Strategy']   
32132      Laush Studio                  ['Indie', 'Racing', 'Simulation']   
32133          SIXNAILS                                ['Casual', 'Indie']   
32134               NaN                                                NaN   

                       app_name                     title  \
0           Lost Summoner Kitty       Lost Summoner Kitty   
1                     Ironbound                 Ironbound   
2       Real Pool 3D - Poolians   Real Pool 3D - Poolians   
3                       弹炸人2222                   弹炸人2222   
4                 Log Challenge                       NaN   
...                         ...                       ...   
32130            Colony On Mars            Colony On Mars   
32131  LOGistICAL: South Africa  LOGistICAL: South Africa   
32132             Russian Roads             Russian Roads   
32133       EXIT 2 - Directions       EXIT 2 - Directions   
32134               Maze Run VR                       NaN   

                                                     url release_date  \
0      http://store.steampowered.com/app/761140/Lost_...   2018-01-04   
1      http://store.steampowered.com/app/643980/Ironb...   2018-01-04   
2      http://store.steampowered.com/app/670290/Real_...   2017-07-24   
3         http://store.steampowered.com/app/767400/2222/   2017-12-07   
4      http://store.steampowered.com/app/773570/Log_C...          NaN   
...                                                  ...          ...   
32130  http://store.steampowered.com/app/773640/Colon...   2018-01-04   
32131  http://store.steampowered.com/app/733530/LOGis...   2018-01-04   
32132  http://store.steampowered.com/app/610660/Russi...   2018-01-04   
32133  http://store.steampowered.com/app/658870/EXIT_...   2017-09-02   
32134  http://store.steampowered.com/app/681550/Maze_...          NaN   

                                                    tags  \
0      ['Strategy', 'Action', 'Indie', 'Casual', 'Sim...   
1      ['Free to Play', 'Strategy', 'Indie', 'RPG', '...   
2      ['Free to Play', 'Simulation', 'Sports', 'Casu...   
3                      ['Action', 'Adventure', 'Casual']   
4                ['Action', 'Indie', 'Casual', 'Sports']   
...                                                  ...   
32130      ['Strategy', 'Indie', 'Casual', 'Simulation']   
32131                    ['Strategy', 'Indie', 'Casual']   
32132                  ['Indie', 'Simulation', 'Racing']   
32133  ['Indie', 'Casual', 'Puzzle', 'Singleplayer', ...   
32134  ['Early Access', 'Adventure', 'Indie', 'Action...   

                                             reviews_url  \
0      http://steamcommunity.com/app/761140/reviews/?...   
1      http://steamcommunity.com/app/643980/reviews/?...   
2      http://steamcommunity.com/app/670290/reviews/?...   
3      http://steamcommunity.com/app/767400/reviews/?...   
4      http://steamcommunity.com/app/773570/reviews/?...   
...                                                  ...   
32130  http://steamcommunity.com/app/773640/reviews/?...   
32131  http://steamcommunity.com/app/733530/reviews/?...   
32132  http://steamcommunity.com/app/610660/reviews/?...   
32133  http://steamcommunity.com/app/658870/reviews/?...   
32134  http://steamcommunity.com/app/681550/reviews/?...   

                         

In [19]:
dfPrueba1.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 32135 entries, 0 to 32134
Data columns (total 13 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   publisher     24073 non-null  object 
 1   genres        28852 non-null  object 
 2   app_name      32133 non-null  object 
 3   title         30085 non-null  object 
 4   url           32135 non-null  object 
 5   release_date  30068 non-null  object 
 6   tags          31972 non-null  object 
 7   reviews_url   32133 non-null  object 
 8   specs         31465 non-null  object 
 9   price         30758 non-null  object 
 10  early_access  32135 non-null  bool   
 11  id            32133 non-null  float64
 12  developer     28836 non-null  object 
dtypes: bool(1), float64(1), object(11)
memory usage: 3.0+ MB


### *2.2. Users_items Dataset:*

In [21]:
df_users_items1 = df_users_items.copy()
df_users_items1

user_id  items_count           steam_id  \
0            76561197970982479          277  76561197970982479   
1                      js41637          888  76561198035864385   
2                    evcentric          137  76561198007712555   
3                   Riot-Punch          328  76561197963445855   
4                        doctr          541  76561198002099482   
...                        ...          ...                ...   
88305        76561198323066619           22  76561198323066619   
88306        76561198326700687          177  76561198326700687   
88307  XxLaughingJackClown77xX            0  76561198328759259   
88308        76561198329548331            7  76561198329548331   
88309        edward_tremethick            0  76561198331598578   

                                                user_url  \
0      http://steamcommunity.com/profiles/76561197970...   
1                   http://steamcommunity.com/id/js41637   
2                 http://steamcommunity.com/id/evcentric   
3                http://steamcommunity.com/id/Riot-Punch   
4                     http://steamcommunity.com/id/doctr   
...                                                  ...   
88305  http://steamcommunity.com/profiles/76561198323...   
88306  http://steamcommunity.com/profiles/76561198326...   
88307  http://steamcommunity.com/id/XxLaughingJackClo...   
88308  http://steamcommunity.com/profiles/76561198329...   
88309     http://steamcommunity.com/id/edward_tremethick   

                                                   items  
0      [{'item_id': '10', 'item_name': 'Counter-Strik...  
1      [{'item_id': '10', 'item_name': 'Counter-Strik...  
2      [{'item_id': '1200', 'item_name': 'Red Orchest...  
3      [{'item_id': '10', 'item_name': 'Counter-Strik...  
4      [{'item_id': '300', 'item_name': 'Day of Defea...  
...                                                  ...  
88305  [{'item_id': '413850', 'item_name': 'CS:GO Pla...  
88306  [{'item_id': '11020', 'item_name': 'TrackMania...  
88307                                                 []  
88308  [{'item_id': '304930', 'item_name': 'Unturned'...  
88309                                                 []  

[88310 rows x 5 columns]

We can see that the column "items" has a list of dictionaries per userid. Let's unnest this column.

In [22]:
# We expand the dataframe users_items in the column "items" in order to convert each element of the lists into one unique row.

df_users_items_Mod_exploded = df_users_items1.explode('items')
df_users_items_Mod_exploded

user_id  items_count           steam_id  \
0      76561197970982479          277  76561197970982479   
0      76561197970982479          277  76561197970982479   
0      76561197970982479          277  76561197970982479   
0      76561197970982479          277  76561197970982479   
0      76561197970982479          277  76561197970982479   
...                  ...          ...                ...   
88308  76561198329548331            7  76561198329548331   
88308  76561198329548331            7  76561198329548331   
88308  76561198329548331            7  76561198329548331   
88308  76561198329548331            7  76561198329548331   
88309  edward_tremethick            0  76561198331598578   

                                                user_url  \
0      http://steamcommunity.com/profiles/76561197970...   
0      http://steamcommunity.com/profiles/76561197970...   
0      http://steamcommunity.com/profiles/76561197970...   
0      http://steamcommunity.com/profiles/76561197970...   
0      http://steamcommunity.com/profiles/76561197970...   
...                                                  ...   
88308  http://steamcommunity.com/profiles/76561198329...   
88308  http://steamcommunity.com/profiles/76561198329...   
88308  http://steamcommunity.com/profiles/76561198329...   
88308  http://steamcommunity.com/profiles/76561198329...   
88309     http://steamcommunity.com/id/edward_tremethick   

                                                   items  
0      {'item_id': '10', 'item_name': 'Counter-Strike...  
0      {'item_id': '20', 'item_name': 'Team Fortress ...  
0      {'item_id': '30', 'item_name': 'Day of Defeat'...  
0      {'item_id': '40', 'item_name': 'Deathmatch Cla...  
0      {'item_id': '50', 'item_name': 'Half-Life: Opp...  
...                                                  ...  
88308  {'item_id': '373330', 'item_name': 'All Is Dus...  
88308  {'item_id': '388490', 'item_name': 'One Way To...  
88308  {'item_id': '521570', 'item_name': 'You Have 1...  
88308  {'item_id': '519140', 'item_name': 'Minds Eyes...  
88309                                                NaN  

[5170015 rows x 5 columns]

In [23]:
# Now we need to convert all the dictionaries of the column "items" through the function json.normalized

df_users_items_Mod_normalized = pd.json_normalize(df_users_items_Mod_exploded['items'])
df_users_items_Mod_normalized

item_id                      item_name  playtime_forever  \
0            10                 Counter-Strike               6.0   
1            20          Team Fortress Classic               0.0   
2            30                  Day of Defeat               7.0   
3            40             Deathmatch Classic               0.0   
4            50      Half-Life: Opposing Force               0.0   
...         ...                            ...               ...   
5170010  373330                    All Is Dust               0.0   
5170011  388490  One Way To Die: Steam Edition               3.0   
5170012  521570          You Have 10 Seconds 2               4.0   
5170013  519140                     Minds Eyes               3.0   
5170014     NaN                            NaN               NaN   

         playtime_2weeks  
0                    0.0  
1                    0.0  
2                    0.0  
3                    0.0  
4                    0.0  
...                  ...  
5170010              0.0  
5170011              3.0  
5170012              4.0  
5170013              3.0  
5170014              NaN  

[5170015 rows x 4 columns]

In [24]:
#Reset the index for each dataframe:

df_users_items_Mod_exploded.reset_index(drop=True, inplace=True)
df_users_items_Mod_normalized.reset_index(drop=True, inplace=True)

In [25]:
#We concatenate both dataframes:
df_users_items_expanded = pd.concat([df_users_items_Mod_exploded.drop('items',axis=1),df_users_items_Mod_normalized], axis=1)
df_users_items_expanded

user_id  items_count           steam_id  \
0        76561197970982479          277  76561197970982479   
1        76561197970982479          277  76561197970982479   
2        76561197970982479          277  76561197970982479   
3        76561197970982479          277  76561197970982479   
4        76561197970982479          277  76561197970982479   
...                    ...          ...                ...   
5170010  76561198329548331            7  76561198329548331   
5170011  76561198329548331            7  76561198329548331   
5170012  76561198329548331            7  76561198329548331   
5170013  76561198329548331            7  76561198329548331   
5170014  edward_tremethick            0  76561198331598578   

                                                  user_url item_id  \
0        http://steamcommunity.com/profiles/76561197970...      10   
1        http://steamcommunity.com/profiles/76561197970...      20   
2        http://steamcommunity.com/profiles/76561197970...      30   
3        http://steamcommunity.com/profiles/76561197970...      40   
4        http://steamcommunity.com/profiles/76561197970...      50   
...                                                    ...     ...   
5170010  http://steamcommunity.com/profiles/76561198329...  373330   
5170011  http://steamcommunity.com/profiles/76561198329...  388490   
5170012  http://steamcommunity.com/profiles/76561198329...  521570   
5170013  http://steamcommunity.com/profiles/76561198329...  519140   
5170014     http://steamcommunity.com/id/edward_tremethick     NaN   

                             item_name  playtime_forever  playtime_2weeks  
0                       Counter-Strike               6.0              0.0  
1                Team Fortress Classic               0.0              0.0  
2                        Day of Defeat               7.0              0.0  
3                   Deathmatch Classic               0.0              0.0  
4            Half-Life: Opposing Force               0.0              0.0  
...                                ...               ...              ...  
5170010                    All Is Dust               0.0              0.0  
5170011  One Way To Die: Steam Edition               3.0              3.0  
5170012          You Have 10 Seconds 2               4.0              4.0  
5170013                     Minds Eyes               3.0              3.0  
5170014                            NaN               NaN              NaN  

[5170015 rows x 8 columns]

In [26]:
df_users_items_expanded.to_csv('users_items_final.csv', index=False)

In [27]:
dfPrueba2 = pd.read_csv('/content/drive/MyDrive/Colab Notebooks/users_items_final.csv')
dfPrueba2

user_id  items_count           steam_id  \
0        76561197970982479          277  76561197970982479   
1        76561197970982479          277  76561197970982479   
2        76561197970982479          277  76561197970982479   
3        76561197970982479          277  76561197970982479   
4        76561197970982479          277  76561197970982479   
...                    ...          ...                ...   
5170010  76561198329548331            7  76561198329548331   
5170011  76561198329548331            7  76561198329548331   
5170012  76561198329548331            7  76561198329548331   
5170013  76561198329548331            7  76561198329548331   
5170014  edward_tremethick            0  76561198331598578   

                                                  user_url   item_id  \
0        http://steamcommunity.com/profiles/76561197970...      10.0   
1        http://steamcommunity.com/profiles/76561197970...      20.0   
2        http://steamcommunity.com/profiles/76561197970...      30.0   
3        http://steamcommunity.com/profiles/76561197970...      40.0   
4        http://steamcommunity.com/profiles/76561197970...      50.0   
...                                                    ...       ...   
5170010  http://steamcommunity.com/profiles/76561198329...  373330.0   
5170011  http://steamcommunity.com/profiles/76561198329...  388490.0   
5170012  http://steamcommunity.com/profiles/76561198329...  521570.0   
5170013  http://steamcommunity.com/profiles/76561198329...  519140.0   
5170014     http://steamcommunity.com/id/edward_tremethick       NaN   

                             item_name  playtime_forever  playtime_2weeks  
0                       Counter-Strike               6.0              0.0  
1                Team Fortress Classic               0.0              0.0  
2                        Day of Defeat               7.0              0.0  
3                   Deathmatch Classic               0.0              0.0  
4            Half-Life: Opposing Force               0.0              0.0  
...                                ...               ...              ...  
5170010                    All Is Dust               0.0              0.0  
5170011  One Way To Die: Steam Edition               3.0              3.0  
5170012          You Have 10 Seconds 2               4.0              4.0  
5170013                     Minds Eyes               3.0              3.0  
5170014                            NaN               NaN              NaN  

[5170015 rows x 8 columns]